hive/agents/qnets/rnn.py

```diff
@@ -0,0 +1,136 @@
+import numpy as np
+import torch
+from torch import nn
+
+from hive.agents.qnets.mlp import MLPNetwork
+from hive.agents.qnets.conv import ConvNetwork
+from hive.agents.qnets.utils import calculate_output_dim
+from hive.agents.qnets.sequence_models import SequenceModule
+
+
+class ConvRNNNetwork(nn.Module):
+    """
+    Basic convolutional recurrent neural network architecture. Applies a number of
+    convolutional layers (each followed by a ReLU activation), recurrent layers, and then
+    feeds the output into an :py:class:`hive.agents.qnets.mlp.MLPNetwork`.
+
+    Note, if :obj:`channels` is :const:`None`, the network created for the
+    convolution portion of the architecture is simply an
+    :py:class:`torch.nn.Identity` module. If :obj:`mlp_layers` is
+    :const:`None`, the mlp portion of the architecture is an
+    :py:class:`torch.nn.Identity` module.
+    """
+
+    def __init__(
+        self,
+        in_dim,
+        sequence_fn: SequenceModule,
```

Comment:
See my comment in the sequence_models.py file about this.

Reply:
done